packages/core/strapi/lib/commands/utils/pkg.js

```diff
@@ -0,0 +1,420 @@
+'use strict';
+
+const fs = require('fs/promises');
+const chalk = require('chalk');
+const pkgUp = require('pkg-up');
+const yup = require('yup');
+
+/**
+ * Utility functions for loading and validating package.json
+ * this includes the specific validation of specific parts of
+ * the package.json.
+ */
+
+/**
+ * The schema for the package.json that we expect,
+ * currently pretty loose.
+ */
+const packageJsonSchema = yup.object({
+  name: yup.string().required(),
+  version: yup.string().required(),
+  type: yup.string().matches(/(commonjs|module)/),
+  license: yup.string(),
+  bin: yup.mixed().oneOf([
+    yup.string(),
+    yup.object({
+      [yup.string()]: yup.string(),
+    }),
+  ]),
+  main: yup.string(),
+  module: yup.string(),
+  source: yup.string(),
+  types: yup.string(),
+  exports: yup.lazy((value) =>
+    yup.object(
+      typeof value === 'object'
+        ? Object.entries(value).reduce((acc, [key, value]) => {
+            if (typeof value === 'object') {
+              acc[key] = yup
+                .object({
+                  types: yup.string(),
+                  source: yup.string(),
+                  import: yup.string(),
+                  require: yup.string(),
+                  default: yup.string(),
+                })
+                .noUnknown(true);
+            } else {
+              acc[key] = yup
+                .string()
+                .matches(/^\.\/.*\.json$/)
+                .required();
+            }
+
+            return acc;
+          }, {})
+        : undefined
+    )
+  ),
+  files: yup.array(yup.string()),
+  scripts: yup.object(),
+  dependencies: yup.object(),
+  devDependencies: yup.object(),
+  peerDependencies: yup.object(),
+  engines: yup.object(),
+});
+
+/**
+ * @typedef {import('yup').Asserts<typeof packageJsonSchema>} PackageJson
+ */
+
+/**
+ * @description being a task to load the package.json starting from the current working directory
+ * using `pkg-up` and `fs` to read the file. If no package.json is found, the process will throw
+ * with an appropriate error message.
+ *
+ * @type {(args: { cwd: string, logger: import('./logger').Logger }) => Promise<object>}
+ */
+const loadPkg = async ({ cwd, logger }) => {
```

Comment:
I don't think we need too, it should find the closest package.json to build your source code from, if you've ran the command in a package of your monorepo it will pick up the package.json of said package. A user doesn't need to run the build command if they have local plugins in their strapi-app & if they're doing a monorepo the command should be ran in the package similar to our own set up 👍🏼

Reply:
You don't change the base dir once you find the package.json I believe so if you aren't in the right folder it won't build correctly any no ?

Reply:
Yep i think you're right actually, lets move to shallow. Also the `strapi` cli won't let you run commands if it's not the root anyway 👍🏼